### DISCUSSION TITLE
How to fill missing values in categorical data?

### DISCUSSION THREAD

**Discussion by GR8xlr8** [community/community#36341] [Other Feature Feedback, Questions, & Ideas] [answered]
I have a dataset of 20000 employees which has following three columns with missing values:

Passing year of College
College specialization
Name of College
Now I have 10000 employees who never went to college. My final aim is to predict their salary.

How can I fill in missing values in this case.

**Comment by arana2k** [comment, marked as answer]
Missing values can be dealt with number of ways, which way to follow depends on the kind of data you have.

Deleting the rows with missing values

Rows with more number of column values as null could be dropped. (Again what is exactly more number depends on individual use case)

Imputing the missing vlaues with Mean / Median

For the numerical Columns you can try replacing the missing values by taking Mean / Median of the column values.

Most frequent Values: Applicable to your Scenario

This method is suitable for Categorical data which i assume is your case. You can try replacing missing vlaues in all three Columns with the most frequently occuring value in the given column.

**Comment by is-it-ayush** [comment]
Considering you're going to predict, I'll assume you're using Python and Some Linear Regression. You might also have the data already in a Pandas Dataframe or as a Numpy Array. So It'll be a lot easier. Just follow the article below.

If not, I'd suggest you export the data set as a `.csv` and load it up in Python as Pandas Dataframe Object. Then you can simply fill in those missing values. This article might help you. [Fill Missing Values with Pandas](https://www.delftstack.com/howto/python-pandas/fill-missing-values-in-pandas/)
